src/react/react.ts

```diff
@@ -385,8 +391,8 @@ class RenderOne<P> implements Subscription {
     const { component, props } = liftedComponent.props
     const renderCache = render(component, props, [value])
 
-    if (!structEq(liftedComponent.state.renderCache, renderCache))
-      liftedComponent.setState({ renderCache })
+    liftedComponent.setState(prevState =>
```

Comment:
Calling `setState(updateFn)` with the same state doesn't re-render component.
Doesn't find anything about that in React documentation.